### PR TITLE
Basic setup

### PR DESCRIPTION
**Pull request by Joozty**
@viktor-yakubiv Not sure if you want to do a review. If not feel free to merge. This PR is just about some basic setup of webpack and installing some necessary npm packages.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 node_modules
+.idea
```
Comment (viktor-yakubiv) [inline .gitignore]: It's better to properly ignore your editor files with [global ignore file](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) and keep `.gitignore` clean

**Comment by viktor-yakubiv** [comment]
We (try to) follow [these commit guidelines](https://chris.beams.io/posts/git-commit/#imperative), so we prefer imperative mode in commit message subject.

File: .gitignore (hunk 1 above)
Comment (Joozty) [reply .gitignore]: @viktor-yakubiv I didn't know about the existence of global ignore file but I still think it's better to keep it in local `.gitignore`. Actually, I would be even  more strict and generate automatic `.gitignore` file for all known IDEs and code related stuff like cache directories  `.npm-cache` ... It prevents all of the people who want to contribute to this repository to accidentally push unwanted stuff not just me. :)

**Comment by Joozty** [comment]
@viktor-yakubiv As I already told you via Slack you are right about webpack. I am going to unify webpack config and create only one file but not in this PR as there are few other PR connected to this branch and I don't want to do rebase or solving some conflicts (I am too lazy for that :smile: ).

**Pull request merged**